### ISSUE TITLE
Unexpected behaviour of tf.keras.activations.relu

### ISSUE THREAD

**Issue by alvaro-garcia-carrasco** [tensorflow/tensorflow#42885]
**System information**
TensorFlow version: v2.3.0-0-gb36436b087 2.3.0

**Describe the current behavior**
When passing `np.nan` to `tf.keras.activations.relu`, it returns 0.0. 
This only happens when using GPU. The relu activation behaves as expected under CPU (it returns `nan`).

**Describe the expected behavior**
When passing `np.nan` to `tf.keras.activations.relu`, it should return `nan`.

**Standalone code to reproduce the issue**
https://colab.research.google.com/drive/1FhsgStZ_3tIIC3WJrCBl3TOjtgWyn2_2?usp=sharing

![image](https://user-images.githubusercontent.com/47107572/91958933-4583ac80-ed08-11ea-95d2-854e631c114e.png)

**Comment by Saduf2019** [comment]
@alvaro-garcia-carrasco 
I ran the code shared on tf-nightly and face a different error, please find the [gist shared](https://colab.research.google.com/gist/Saduf2019/341f62d0d325f5863ac2e545f1177504/untitled401.ipynb).

**Comment by alvaro-garcia-carrasco** [comment]
@Saduf2019 
The bug seems to be fixed in the tf-nightly version. 

I don't know how to check that I am actually running on GPU in the tf-nightly version as the code I used to check for that in tf-2.3.0 now raises an error: 
```assert len(tf.config.list_physical_devices('GPU')) > 0``` 

But the important thing is that in the nightly version the relu activation returns a `nan` when passed a `nan`.

**Comment by bhack** [comment]
I am not sure that It Is solved in nightly:
See https://github.com/tensorflow/tensorflow/issues/42957

**Comment by bhack** [comment]
Moreover I think that this could be closed as a duplicate of https://github.com/tensorflow/tensorflow/issues/40072

**Comment by bhack** [comment]
@Saduf2019 It seems that you was the triager also on June mentioned ticket.

**Comment by bhack** [comment]
To confirm https://github.com/tensorflow/tensorflow/issues/42957 claim it is not solved. 

@Saduf2019 if you check it is just a false positive:
`pip install tf-nightly==2.4.0.dev20200818`
`pip install tf-nightly==2.4.0.dev20200819`

**Comment by alvaro-garcia-carrasco** [comment]
This issue is about the unexpected behaviour of tf.keras.activations.relu. This is fixed in the nightly version. 

Unfortunately, the code I provided to test it contained another issue which is addressed elsewhere ( #42957). I would consider this issue closed.

**Comment by bhack** [comment]
@alvaro-garcia-carrasco I don't think so. We can close this just cause it is a duplicated of #40072 already triaged by @Saduf2019 in June.

I don't know if it is limited to Colab instances but you can double test yourself in your machine:
with 
` pip install tf-nightly==2.4.0.dev20200818`
and
`  pip install tf-nightly==2.4.0.dev20200819` or newer
```
import tensorflow as tf
import numpy as np
from tensorflow.python import pywrap_tfe
tf.debugging.set_log_device_placement(True)
sample = np.array([[np.nan]], dtype=np.float32)
x=tf.keras.activations.relu(sample).numpy()
print(x.item())
```

**Comment by Saduf2019** [comment]
> This issue is about the unexpected behaviour of tf.keras.activations.relu. This is fixed in the nightly version.
> 
> Unfortunately, the code I provided to test it contained another issue which is addressed elsewhere ( #42957). I would consider this issue closed.

Thank you for your update, Glad the issue is resolved in tf-nightly. In case of any new issue please feel free to create a new issue.
Moving this issue to closed status with confirmation.

**Comment by bhack** [comment]
@Saduf2019 It Is not solved
